tests/tests/swfs/avm2/geom_transform/Test.as

```diff
@@ -0,0 +1,195 @@
+
+            test3D();
+            trace("");
+
+            testCopy2D();
+            trace("");
+
+            testCopy3D();
+            trace("");
+
+            testImageComparison();
+        }
+
+        private function test2D() : void {
+            var sprite2D : Sprite = new Sprite();
+
+            trace("// sprite2D: new Sprite has null matrix3D and valid matrix");
+            trace("sprite2D.transform.matrix", sprite2D.transform.matrix);
+            trace("sprite2D.transform.matrix3D", sprite2D.transform.matrix3D);
+
+            trace("// sprite2D: set identity matrix");
+            var mat2D : Matrix = new Matrix();
+            mat2D.identity();
+            sprite2D.transform.matrix = mat2D;
+            trace("sprite2D.transform.matrix", sprite2D.transform.matrix);
+            trace("sprite2D.transform.matrix3D", sprite2D.transform.matrix3D);
+            trace("mat2D", mat2D);
+
+            trace("// sprite2D: .matrix = null");
+            sprite2D.transform.matrix = null;
+            trace("sprite2D.transform.matrix", sprite2D.transform.matrix);
+            trace("sprite2D.transform.matrix3D", sprite2D.transform.matrix3D);
+            trace("sprite2D.transform.matrix3D.rawData", sprite2D.transform.matrix3D.rawData);
+            trace("mat2D", mat2D);
+
+            trace("// sprite2D: .matrix3D = null");
+            sprite2D.transform.matrix3D = null;
+            trace("sprite2D.transform.matrix", sprite2D.transform.matrix);
+            trace("sprite2D.transform.matrix3D", sprite2D.transform.matrix3D);
+            trace("mat2D", mat2D);
+
+            trace("// sprite2D: set x = 30, y = 50");
+            sprite2D.x = 30;
+            sprite2D.y = 50;
+            trace("sprite2D.transform.matrix", sprite2D.transform.matrix);
+            trace("sprite2D.transform.matrix3D", sprite2D.transform.matrix3D);
+            trace("mat2D", mat2D);
+        }
+
+        private function test3D() : void {
+            var sprite3D : Sprite = new Sprite();
+
+            trace("// sprite3D: set identity matrix3D");
+            var mat3D : Matrix3D = new Matrix3D();
+            mat3D.identity();
+            sprite3D.transform.matrix3D = mat3D;
+            trace("sprite3D.transform.matrix", sprite3D.transform.matrix);
+            trace("sprite3D.transform.matrix3D", sprite3D.transform.matrix3D);
+            trace("sprite3D.transform.matrix3D.rawData", sprite3D.transform.matrix3D.rawData);
+            trace("mat3D.rawData", mat3D.rawData);
+
+            trace("// sprite3D: .matrix = null");
+            sprite3D.transform.matrix = null;
+            trace("sprite3D.transform.matrix", sprite3D.transform.matrix);
+            trace("sprite3D.transform.matrix3D", sprite3D.transform.matrix3D);
+            trace("sprite3D.transform.matrix3D.rawData", sprite3D.transform.matrix3D.rawData);
+            trace("mat3D.rawData", mat3D.rawData);
+
+            //// FIXME: matrix3D.rawData should be updated by x/y update.
```

Comment:
I believe that's already the case, it's just that we don't reset tx,ty on `matrix3D = null` and the assertions below fail.

We could also do some more assertions related to non-indentity 3D matrices, I think we can safely do that without modifying the code

Reply:
Thank you. You're right. I added more test cases for non-identity matrices plus uncomment: 1b6802d268fa4d236550be7c656ade42694fdee2

And thanks to the test case, I found another interesting behavior. When setting `.transform.matrix3D = null`, the matrix gets reset to the identity matrix. Added a fix for that: 019476db8d210f43d3ab2bac84f1a2b1317d80f4